### PR TITLE
feat(rich-text-editor) - rich style feature

### PR DESCRIPTION
**Pull request by gavin-everett-genesys**
**Option 3 (Chosen Solution)**
Create a new component, gux-rte-list, that mirrors gux-list.
Pros: Provides flexibility, especially for adding other components like a color-picker list to the text-editor in the future.

Also one change I did make is that I extracted the styles from `gux-list-item.scss` and created a new file with a mixin which can be imported into any of the new `list-item` components that will be created.

### CODE REVIEW

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        <button
+          type="button"
+          ref={el => (this.actionButton = el)}
+          class={{ 'gux-is-active': this.expanded }}
+          onClick={() => this.onActionButtonClick()}
+          disabled={this.disabled}
+          aria-haspopup="true"
+          aria-expanded={this.expanded.toString()}
+        >
+          {this.i18n('paragraph')}
```
Comment (gavin-everett-genesys) [inline packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx:160]: Going to confirm with UX if this needs to be dynamic depending on the text-style selected.

**Comment by gavin-everett-genesys** [comment]
Will add tests once I get some feedback on additions.

**Comment by 321gillian** [comment]
new list approach makes sense to me.

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html [hunk 2]
```diff
@@ -9,6 +9,20 @@ <h1>gux-rich-text-editor</h1>
     <gux-rich-text-editor-action action="strike"></gux-rich-text-editor-action>
   </gux-rich-text-editor-action-group>
   <gux-rich-text-editor-action-group slot="text-styling">
+    <gux-rich-text-editor-action-rich-style>
+      <gux-rich-style-list-item text-style="heading-1"
+        >Heading 1</gux-rich-style-list-item
```
[the remark this answers is not in the document]
Comment (gavin-everett-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html:14]: Yeah thats a fair point, I think when I was creating it I was thinking that there might be a possibility they would want to change the text for some reason. I will confirm with UX but yeah it should probably be handled by the component.

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  actionButton: HTMLElement;
+  listElement: HTMLGuxRichTextEditorListElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  @Prop({ mutable: true })
+  expanded: boolean = false;
+
+  @Prop({ mutable: true })
```
[the remark this answers is not in the document]
Comment (gavin-everett-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx:30]: Ye I am mutating it in the `componentWillLoad` method

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+export class GuxRichTextEditorActionRichStyle {
+  private i18n: GetI18nValue;
+  actionButton: HTMLElement;
+  listElement: HTMLGuxRichTextEditorListElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  @Prop({ mutable: true })
+  expanded: boolean = false;
```
[the remark this answers is not in the document]
Comment (gavin-everett-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx:28]: Good catch will change to `State`

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  private focusLastListItem(): void {
+    afterNextRender(() => {
+      void this.listElement.guxFocusLastItem();
+    });
+  }
+
+  async componentWillLoad(): Promise<void> {
+    trackComponent(this.root);
+    this.i18n = await buildI18nForComponent(this.root, translationResources);
+    this.disabled = calculateDisabledState(this.root);
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx:116]: I don't think this is the right thing to do. Toggling the parents disabled state will change this items disabled state permanently.
If this item is disabled and I toggle the parent from enabled to disabled to enabled this item should still be disabled.
If this does not make sense I can go on zoom to explain my issue better.

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (gavin-everett-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html:14]: Confirmed with UX that the text should be slotted.

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx (hunk 5 above)
Comment (gavin-everett-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx]: That makes sense to me 👍  If it is set here it wont change due it only executing once. I think its probably better off attached to the class maybe. I have updated the change @daragh-king-genesys

File: packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (daragh-king-genesys) [reply packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/example.html:14]: Had a chat with @gavin-everett-genesys on Zoom and he is going to have another look at the api here.
He is going to confirm if UX want the list of possible values to be limited to what we provide or if apps should be able to add extra items to the list. Depending on what UX wants the best api will be a little different.

**Comment by gavin-everett-genesys** [comment]
This PR is now ready for review and the items we discussed in last Fridays collab time have been implemented.

**Review by jordanstith15 — APPROVED** [review]

**Pull request merged**